cloud/shared/bin/run

```diff
@@ -13,6 +13,8 @@
 # install the package and activate the venv. The activation has to happen in the
 # parent process before the python code is run.
 # This is why we need a bash script that preceeds the python code
+set -e
+set -o pipefail
```

Comment:
**suggestion (non-blocking):** we should move toward using the same pattern in these shell scripts that we do in the main repo `bin/` scripts wherein each script begins with `source lib.sh` -- in this case it'd be `source cloud/shared/bin/lib.sh` which sets standard options like these and loads shared code e.g. below where you do `echo "Latest snapshot tag not found." 2>&1` could be `out::error "Latest snapshot tag not found."`

That's a little off-topic for this PR so feel free to merge as-is.

Reply:
oo good point! i'll update this.

Reply:
I ran into some problems refactoring the shell scripts so I'm leaving this as is for now and would love to pair on this with you next week sometime!